### PR TITLE
Adding package exporting injection tokens for cluster settings 

### PR DESCRIPTION
**Pull request by aleksfront**
Third part of https://github.com/lensapp/lens/issues/3846. Second part is https://github.com/lensapp/lens/pull/7365.

Adding `cluster-settings` inner package which exports injection tokens needed for cluster settings icon configuration.

### CODE REVIEW

File: packages/cluster-settings/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "types": "./dist/index.d.ts",
+  "files": [
+    "dist"
+  ],
+  "scripts": {
+    "clean": "rimraf dist/",
+    "generate-types": "tsc --d --declarationDir ./dist --declarationMap --emitDeclarationOnly",
+    "build": "npm run generate-types && swc ./src/index.ts -d ./dist"
+  },
+  "devDependencies": {
```
Comment (ixrock) [inline packages/cluster-settings/package.json:23]: Should we include `typescript` as dev dependency as well?

File: packages/core/src/renderer/components/cluster-settings/__tests__/icon-settings.test.tsx [hunk 2]
```diff
@@ -53,6 +56,29 @@ const newMenuItem = getInjectable({
   injectionToken: clusterIconSettingsMenuInjectionToken,
 });
 
+function customSettingsComponent(props: ClusterIconSettingComponentProps) {
```
Comment (ixrock) [inline packages/core/src/renderer/components/cluster-settings/__tests__/icon-settings.test.tsx]: _nit:_ 

```suggestion
function CustomSettingsComponent(props: ClusterIconSettingComponentProps) {
```

File: packages/core/src/renderer/components/cluster-settings/__tests__/icon-settings.test.tsx [hunk 3]
```diff
@@ -39,6 +40,8 @@ const clusterEntity = new KubernetesCluster({
   },
 });
 
+cluster.preferences.clusterName = "some-cluster-name";
```
Comment (ixrock) [inline packages/core/src/renderer/components/cluster-settings/__tests__/icon-settings.test.tsx]: Looking at `ClusterModel` (first argument of `Cluster`) it seems you can pass `preferences` in the constructor above.

```ts
export interface ClusterModel {
  /** Unique id for a cluster */
  id: ClusterId;

  /** Path to cluster kubeconfig */
  kubeConfigPath: string;

  /** User context in kubeconfig  */
  contextName: string;

  /** Preferences */
  preferences?: ClusterPreferences;

  /** Metadata */
  metadata?: ClusterMetadata;

  /** List of accessible namespaces */
  accessibleNamespaces?: string[];

  /**
   * Labels for the catalog entity
   */
  labels?: Partial<Record<string, string>>;
}
```

**Review by ixrock — DISMISSED** [review]

File: packages/cluster-settings/package.json (hunk 1 above)
Comment (aleksfront) [reply packages/cluster-settings/package.json:23]: Not sure. It works well without specify typescript here. Probably it uses some globally installed typescript somehow.

Comment (ixrock) [reply packages/cluster-settings/package.json:23]: Yes, try to remove global ‘typescript’ package and check again.

**Review by ixrock — APPROVED** [review]
LGTM

File: packages/cluster-settings/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@k8slens/cluster-settings",
+  "version": "6.5.0-alpha.1",
```
Comment (Nokel81) [inline packages/cluster-settings/package.json:3]: Nit: this should probably be `1.0.0-alpha.1`

Comment (aleksfront) [reply packages/cluster-settings/package.json:3]: Following same versioning as other internal packages. Keeping `6.5.0-alpha.1` for now until clarification. Will change it if needed in follow up PR.

**Pull request merged**